lib/metrics.rb

```diff
@@ -0,0 +1,183 @@
+# frozen_string_literal: true
+
+module Metrics
+  class TimeSeries
+    attr_accessor :labels, :query
+
+    def initialize(labels:, query:)
+      @labels = labels
+      @query = query
+    end
+  end
+
+  class MetricDefinition
+    attr_accessor :name, :description, :unit, :series
+
+    def initialize(name:, description:, unit:, series:)
+      @name = name
+      @description = description
+      @unit = unit
+      @series = series
+    end
+  end
```

Comment:
You can use the Data core class. It's similar to a struct but immutable and also allow named arguments.

https://www.shakacode.com/blog/ruby-3-2-adds-a-new-data-class/

Reply:
Nice, I was looking for something similar to Python's `dataclasses`, this looks like it

Reply:
Done